src/angular/hq/src/app/clients/client-edit/client-edit.component.ts

```diff
@@ -0,0 +1,112 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { HQService } from '../../services/hq.service';
+import { firstValueFrom } from 'rxjs';
+import { ActivatedRoute, Router } from '@angular/router';
+import { APIError } from '../../errors/apierror';
+
+
+interface Form {
+  name: FormControl<string>;
+  officialName: FormControl<string | null>;
+  billingEmail: FormControl<string | null>;
+  hourlyRate: FormControl<number | null>;
+}
+
+@Component({
+  selector: 'hq-client-edit',
+  standalone: true,
+  imports: [CommonModule, ReactiveFormsModule],
+  templateUrl: './client-edit.component.html'
+})
+
+export class ClientEditComponent implements OnInit{
+  clientId?: string;
+  ngOnInit() {
+    this.route.paramMap.subscribe(params => {
+      this.clientId = params.get('clientId') ?? undefined
+      console.log(this.clientId)
+      this.getClient();
+    })
```

Comment:
Instead of subscribing here (Which has the potential to emit multiple values if the route changes), use `firstValueFrom` with something like this:

```
this.clientId = await firstValueFrom(this.route.paramsMap.map(params.get('clientId')));
this.getClient();
```

ngOnInit/ngOnDestroy/etc. can be made async and Angular will appropraitely await any Promises that are returned.